### PR TITLE
MSCI 7/8/9: Handle Account creation

### PR DESCRIPTION
**Pull request by abcotter**
**In this PR:**

1. The docker-compose file was adjusted to incorporate the DB initialization, the "Users" table is created upon spin up to have 3 columns: userId (autoincrement), userName, userPassword

2. The python API endpoint was written to assert username uniqueness and throw an error if not unique. and also add new users if possible.

3. Build out two vue components:
- "create account" button that simply routes to the create account page (to make incorporating into Mo's code more seamless) 
- create account page!

If you have feebback for if the account creation should log them in or route them back to the login page let me know!

**Testing:** I tested my code by running each edge case: incomplete form, non-matching passwords, non-unique username and unique username. all cases behaved as expected.

**Important notes!!** 
This ticket I have added a new library for the notification display handling, if you have errors locally you may need to run:
npm install --save vue-notification

for more information on how to use the new notification library please refer to: [here](https://github.com/euvl/vue-notification/blob/master/README.md)

### CODE REVIEW

**Review by CarlaLeal — COMMENTED** [review]
I recommend creating  a custom context manager that sets up and closes the connections/cursors to avoid code duplication
```
class DbSelector():
    def __init__(self):
        self.config = {'user': ',
                       'password': '',
                       'host': ,
                       'port': ,
                       'database':'}
    def __enter__(self):
        self.connection = mysql.connector.connect(**self.config)
        self.cursor = self.connection.cursor(prepared=True)
        return self
    def __exit__(self, exc_type, exc_value, exc_traceback):
        self.connection.commit()
        self.connection.close()
        self.cursor.close()

**Review by CarlaLeal — CHANGES_REQUESTED** [review]
Can we use environment variables for the mysql config so that when we deploy to production it reads the right config?

File: app/app.py [hunk 1]
```diff
@@ -1,53 +1,94 @@
...
+    cursor = connection.cursor()
+
+    query = "INSERT INTO Users (userName, userPassword) VALUES (%s, %s);"
+
+    cursor.execute(query, [username, password])
+    connection.commit()
+    connection.close()
+    cursor.close()
+
+    print(cursor)
```
Comment (CarlaLeal) [inline app/app.py]: Could we remove print statements?

**Comment by abcotter** [comment]
@carla I reviewed you comments and agree, I've incorporated you feedback! thanks!

**Comment by abcotter** [comment]
@CarlaLeal I've made the changes requested above! I also merged in main to get the new Production code. I've made changes and it's working locally. 

I've added the ENV variables to the docker-compose.yml file for the local variables to access the DB can you review my changes to the app.py file to use env variables and confirm this will work with production/ heroku?

thanks!

**Review by CarlaLeal — APPROVED** [review]

**Pull request merged**